docs/content/Schema/Dynamic-Schema-Creation.md

```diff
@@ -0,0 +1,122 @@
+---
+title: Dynamic Schema Creation
+permalink: /schema/dynamic-schema-creation
+category: Data Schema
+menuOrder: 2
+---
+
+Cube.js allows schemas to be created on-the-fly using a special
+[`asyncModule()`][ref-async-module] function only available in the [schema
+execution environment][ref-schema-env]. `asyncModule()` allows registering an
+async function to be executed at the end of the data schema compile phase so
+additional definitions can be added.
+
+[ref-schema-env]: /schema-execution-environment
+[ref-async-module]: /schema-execution-environment#asyncmodule
+
+This is often useful in situations where schema properties can be dynamically
+updated through an API, for example.
+
+Some example scenarios are below:
+
+<!-- prettier-ignore-start -->
+[[warning | Note]]
+| Each `asyncModule` call will be invoked only once per schema compilation.
+<!-- prettier-ignore-end -->
+
+## Generation
+
+In the following example, we retrieve a JSON object representing all our cubes
+using `fetch()` and then use the [`cube()` global function][ref-globals] to
+generate schemas from that data:
+
+[ref-globals]:
+  https://cube.dev/docs/schema-execution-environment#cube-js-globals-cube-and-others
+
+```javascript
+// schema/DynamicSchema.js
+const fetch = require('node-fetch');
+
+asyncModule(async () => {
+  const dynamicCubes = await (
+    await fetch('http://your-api-endpoint/dynamicCubes')
+  ).json();
```

Comment:
Can we show an example of how this JSON should look like? So users would know how to format `/dynamicCubes`API.